run_custom_code_jobs.py

```diff
@@ -0,0 +1,407 @@
+        files = list_repo_files(repo_id, repo_type=repo_type)
+        return set(files)
+    except Exception as e:
+        logger.warning(f"Failed to list files in {repo_id}: {e}")
+        return set()
+
+# === Notebook Processing ===
+
+def fetch_notebook_content(model_id: str) -> Optional[Dict]:
+    """
+    Fetch Jupyter notebook from HuggingFace model repository.
+    
+    Args:
+        model_id: HuggingFace model identifier
+        
+    Returns:
+        Parsed notebook content or None if fetch fails
+    """
+    notebook_url = f"https://huggingface.co/{model_id}.ipynb"
+    
+    try:
+        response = requests.get(notebook_url, timeout=30)
+        response.raise_for_status()
+        return json.loads(response.text)
+    except requests.exceptions.RequestException as e:
+        logger.error(f"Failed to fetch notebook for {model_id}: {e}")
+        return None
+    except json.JSONDecodeError as e:
+        logger.error(f"Invalid JSON in notebook for {model_id}: {e}")
+        return None
+
+def extract_code_cells(notebook: Dict) -> List[str]:
+    """Extract code cells from Jupyter notebook that start with #."""
+    code_snippets = []
+    
+    for cell in notebook.get("cells", []):
+        if cell.get("cell_type") != "code":
+            continue
+            
+        source_lines = cell.get("source", [])
+        cell_content = "".join(source_lines).strip()
+        
+        # Only process cells that start with # (likely examples or documentation)
+        if cell_content and cell_content.startswith("#"):
+            code_snippets.append(cell_content)
+    
+    return code_snippets
+
+def wrap_code_with_error_handling(code_content: str, model_name: str, snippet_index: int) -> str:
+    """
+    Wrap code snippet with try-except block for error logging.
+    
+    Args:
+        code_content: Original code snippet
+        model_name: Sanitized model name
+        snippet_index: Index of the snippet
+        
+    Returns:
+        Code wrapped with error handling
+    """
+    exception_filename = f"{model_name}_{snippet_index}_exception.txt"
+    timestamp = get_current_timestamp()
+    
+    wrapped_lines = ["try:"]
+    
+    # Indent original code
+    for line in code_content.splitlines():
+        wrapped_lines.append(f"    {line}")
+    
+    # Add exception handling
+    wrapped_lines.extend([
+        "except Exception as e:",
+        f"    exception_file = '{exception_filename}'",
+        "    with open(exception_file, 'w') as f:",
+        "        import traceback",
+        "        traceback.print_exc(file=f)",
+        "    # Upload exception log to HuggingFace",
+        "    from huggingface_hub import upload_file",
+        "    upload_file(",
+        f"        path_or_fileobj=exception_file,",
+        f"        repo_id='{HF_DATASET_EXCEPTIONS_REPO}',",
+        f"        path_in_repo='{timestamp}/{exception_filename}',",
+        "        repo_type='dataset',",
+        "    )"
+    ])
```

Comment:
This is the main part where I am wrapping the code with try catch, so that I could write the exception to a dataset and then figure out what went wrong.